### PR TITLE
Adding custom group styles function

### PR DESCRIPTION
**Pull request by rcdvl**
# Submit a pull request

### 🎯 Goal

Adding a way to change the group styles makes it easier to customize its experience (like grouping only messages within a timeframe, etc).

### 🛠 Implementation details

_Provide a description of the implementation_

### 🎨 UI Changes

_Add relevant screenshots_

### CODE REVIEW

File: src/components/MessageList/hooks/useEnrichedMessages.ts [hunk 1]
```diff
@@ -72,10 +79,11 @@ export const useEnrichedMessages = <
     messagesWithDates = insertIntro(messagesWithDates, headerPosition);
   }
 
+  const groupStylesFn = groupStyles ? groupStyles : getGroupStyles;
```
Comment (ambaldwin) [inline src/components/MessageList/hooks/useEnrichedMessages.ts:82]: Change this to: `const groupStylesFn = groupStyles || getGroupStyles;` to match other overrides in the library.

File: src/components/MessageList/MessageList.tsx [hunk 2]
```diff
@@ -38,6 +38,7 @@ import type {
   DefaultReactionType,
   DefaultUserType,
 } from '../../types/types';
+import type { GroupStyle } from './utils';
```
Comment (ambaldwin) [inline src/components/MessageList/MessageList.tsx:41]: Let's move this import to line 28, above the MessageProp type and with a line in between.

**Review by ambaldwin — COMMENTED** [review]
Thanks for the addition! Please see very minor comments before we approve. We also need to export * from './utils' in MessageList/index.ts so folks can correctly type the return value of the callback function.

**Review by rcdvl — COMMENTED** [review]
@ambaldwin All comments addressed (I think 😄), feel free to make any additional suggestion.

**Review by petyosi — APPROVED** [review]

**Pull request merged**